x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/LongRangeBlockBuilder.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.GenericNamedWriteable;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.index.mapper.BlockLoader;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.mapper.RangeFieldMapper.ESQL_LONG_RANGES;
+
+public class LongRangeBlockBuilder extends AbstractBlockBuilder implements BlockLoader.LongRangeBuilder {
+
+    private LongBlockBuilder fromBuilder;
+    private LongBlockBuilder toBuilder;
+
+    public LongRangeBlockBuilder(int estimatedSize, BlockFactory blockFactory) {
+        super(blockFactory);
+        fromBuilder = null;
+        toBuilder = null;
+        try {
+            fromBuilder = new LongBlockBuilder(estimatedSize, blockFactory);
+            toBuilder = new LongBlockBuilder(estimatedSize, blockFactory);
+        } finally {
+            if (toBuilder == null) {
+                Releasables.closeWhileHandlingException(fromBuilder);
+            }
+        }
+    }
+
+    @Override
+    protected int valuesLength() {
+        throw new UnsupportedOperationException("Not available on date_range");
+    }
+
+    @Override
+    protected void growValuesArray(int newSize) {
+        throw new UnsupportedOperationException("Not available on date_range");
+    }
+
+    @Override
+    protected int elementSize() {
+        throw new UnsupportedOperationException("Not available on date_range");
+    }
+
+    @Override
+    public long estimatedBytes() {
+        return fromBuilder.estimatedBytes() + toBuilder.estimatedBytes();
+    }
+
+    @Override
+    public LongRangeBlockBuilder copyFrom(Block b, int beginInclusive, int endExclusive) {
+        Block fromBlock;
+        Block toBlock;
+        if (b.areAllValuesNull()) {
+            fromBlock = b;
+            toBlock = b;
+        } else {
+            var block = (LongRangeArrayBlock) b;
+            fromBlock = block.getFromBlock();
+            toBlock = block.getToBlock();
+        }
+        fromBuilder.copyFrom(fromBlock, beginInclusive, endExclusive);
+        toBuilder.copyFrom(toBlock, beginInclusive, endExclusive);
+        return this;
+    }
+
+    public LongRangeBlockBuilder copyFrom(LongRangeBlock block, int pos) {
```

Comment:
This is only used in tests. Should it really exist? I wonder about the value of tests that only test a method that is not used anywhere outside the tests?